docs/pages/access-controls/reference.mdx

```diff
@@ -373,3 +374,163 @@ Here is an explanation of the fields used in the `where` and `filter` conditions
 
 Check out our [predicate language](../reference/predicate-language.mdx#scoping-allowdeny-rules-in-role-resources)
 guide for a more in depth explanation of the language.
+
+## Template variables and functions
+
+Teleport provides several template variables and functions to enable more robust access control:
+
+<Details title="Internal Variables and Functions">
+
+#### `{{email.local(variable)}}`
+
+- **Type**: Function
+- **Description**: Extracts the local part of an email address.
+- **Example**: `{{email.local(alice@example.com)}}` -> alice
+
+#### `{{regexp.match(expression, variable)}}`
+
+- **Type**: Function
+- **Description**: Returns true if the provided string matches the specified pattern.
+- **Example**: `{{regexp.match("^Email: (.*)$", external.raw_email)}}` -> true for matching input
+
+#### `{{regexp.not_match(expression, variable)}}`
+
+- **Type**: Function
+- **Description**: Returns true if the provided string does not match the specified pattern.
+- **Example**: `{{regexp.not_match("^Email: (.*)$", external.raw_email)}}` -> true for non-matching input
+
+#### `{{regexp.replace(variable, expression, replacement)}}`
+
+- **Type**: Function
+- **Description**: Finds all matches of the expression and replaces them with the replacement. Supports expansion.
+- **Example**: `{{regexp.replace(external.email, "^(.*)@example.com$", "$1")}}` -> alice (when input is alice@example.com)
+
+#### `{{internal.azure_identities}}`
+
+- **Type**: Variable
+
+#### `{{internal.db_names}}`
+
+- **Type**: Variable
+- **Description**: List of all allowed database names.
+- **Example**: `{{internal.db_names}}` -> ["db_name_1", "db_name_2"]
+
+#### `{{internal.db_users}}`
+
+- **Type**: Variable
+- **Description**: List of all allowed database users.
+- **Example**: `{{internal.db_users}}` -> ["db_user_1", "db_user_2"]
+
+#### `{{internal.gcp_service_accounts}}`
+
+- **Type**: Variable
+
+#### `{{internal.jwt}}`
```

Comment:
I believe the `{{internal.jwt}}` variable is really only valid in the context of the templating done in application access, and not in a role template.

Reply:
Noted below.

Reply:
this is not part of a role template, it is not part of a role at all, so I don't think it should be on this page